1_prep/src/repair_obs_spatial.R

```diff
@@ -0,0 +1,140 @@
+#' Restore spatial information to observed WQP data using WQP
+#' crosswalks and obs data
+#'
+#' @param data data.frame, observed data (`merged_temp_data_daily.feather`) from `lake-temperature-model-prep`
+#' @param xwalk chr, full path and file name of the `sf` crosswalk
+#'
+repair_wqp_data <- function(data, xwalk) {
+
+  sample_sites <- unique(data$source) %>%
+    str_subset(., '^wqp', negate = FALSE)
+
+  df_xwalk <- readRDS(xwalk) %>%
+    # create and mutate a column for matching to data
+    rename(source = site_id) %>%
+    mutate(source = sprintf('wqp_%s', source)) %>%
+    select(- OrganizationIdentifier, - resultCount) %>%
+    filter(source %in% sample_sites)
+
+  out <- left_join(data, df_xwalk) %>%
+    filter(source %in% sample_sites) %>%
+    st_as_sf()
+
+  out_sf <- st_as_sf(out)
+
+  return(out_sf)
+}
+
+#' Create an `sf` object from a csv that contains stations and lat/long
+#'
+#' @param file_in chr, file name and full path for the crosswalk of interest.
+#' @param path_out chr, full path for the location of the `sf` crosswalk
+#' @param transform num, EPSG number that should be used to reproject the crosswalk. The default projection is EPSG 4326.
+#'
+create_missing_xwalk <- function(file_in, path_out, transform = NULL) {
+
+  xwalk_file_out <- file_in %>%
+    basename(.) %>%
+    str_extract(., pattern = '[^.]+') %>%
+    sprintf('%s/%s_sf.rds', path_out, .)
+
+  xwalk_out <- readr::read_csv(file_in, skip = 1, col_types = cols()) %>%
+    select(lat, long, contains('site')) %>%
+    # ESPG for UTM 15 N (Wapapello is just west of the boundary) is 32615
+    st_as_sf(coords = c('long', 'lat'), crs = 4326)
+
+  if(!is.null(transform)) {
+    xwalk_out %>%
+      st_transform(transform) %>%
+      saveRDS(., xwalk_file_out)
+  } else {
+    xwalk_out %>%
+      saveRDS(., xwalk_file_out)
+  }
+
+  return(xwalk_file_out)
+}
+
+#' Restore spatial information to observed cooperator data using
+#' crosswalks and obs data
+#'
+#' @param tbl_row tibble with two rows: one specifying the full file name for the `tmp` data and one for the crosswalk
+#' @param data data.frame, observed data (`merged_temp_data_daily.feather`) from `lake-temperature-model-prep`
+#'
+repair_coop_data <- function(tbl_row, data) {
+
+  # read in data from tibble
+  dat_temp <- readRDS(tbl_row$temp_files)
+  dat_spatial <- readRDS(tbl_row$xwalk_files)
+
+  # name repair for various joins
+  dat_temp <- validate_names(dat_temp, type = 'temp')
+  dat_spatial <- validate_names(dat_spatial, type = 'spatial')
+
+  raw_data_joined <- left_join(dat_temp, dat_spatial) %>%
```

Comment:
I resolved this in this commit: 1407a91c29e60aac43474cf2122b6349d733d9db

I also added in a filter at the end because I found that there were stations with empty geometry that made it through the match.